### PR TITLE
refactor remove default case handling fiserv

### LINKED ISSUE #4059: [REFACTOR]: [Fiserv] Remove Default Case Handling
### :memo: Feature Description

- We utilize match statements to make pivotal decisions, such as generating requests based on the payment method type and managing responses received from the connector. 
- These conditions generally go hand in hand with enum variants. 
- Default case is used because a match statement needs to be exhaustive i.e. every variant needs to be covered. 
- So, if all the explicit cases are handled then default is used to handle the rest.
- Each connector have these match statements but many of them don’t provide reference to each variant in their default case, rather a `_` is put to handle all the other cases. 
- This approach carries a risk because developers may inadvertently overlook the need for explicit handling of the new cases.

### :hammer: Possible Implementation

- Instead of relying on a default match case `_`, developers should handle each and every variant explicitly. 
- By doing so, if there are any changes in the future, they can readily assess the impact of their modifications simply by compiling the code.
- In order to manage payment methods that are not implemented by Hyperswitch or yet to be implemented, the connector transformers file should make use of the NotImplemented ConnectorError enum variant.
- By doing so, we will throw same error message for all the Connector Implementation
- You can check this PR for further reference https://github.com/juspay/hyperswitch/pull/1955

:bookmark: Note: All the changes needed should be contained within `hyperswitch/crates/router/src/connector/fiserv/transformer.rs`

### :package: Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### :package: Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by ind1goDusk**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

this pull request removes default case handling in `crates/router/src/connector/fiserv/transformers.rs` and handles each case explicitly instead. 

## Motivation and Context

closes #4059

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: hyperswitch
Comment (SanchithHegde) [inline hyperswitch:1]: Hey @ind1goDusk, could you push the changes made to the repository, instead of adding a submodule?

**Review by SanchithHegde — COMMENTED** [review]
Also, we would prefer you commenting on the linked issue, so that we can assign it to you.

**Comment by ind1goDusk** [comment]
if this issue is still available, i'd like to be assidned to it. i'm closing this pull request to start from a fresh branch.

**Comment by SanchithHegde** [comment]
> if this issue is still available, i'd like to be assidned to it. i'm closing this pull request to start from a fresh branch.

Yes, @ind1goDusk, the issue is still available. Can you comment on the linked issue so that we can assign it to you?

**Pull request closed without merging**